### PR TITLE
Use Proxy for Teams Notification

### LINKED ISSUE #2588: Notifications via a proxy (squid)
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Telegram (axios.get) and teams (axios.post) use neither the system-wide proxy (HTTP_PROXY, HTTPS_PROXY, NO_PROXY) nor a separately defined one such as in the .npmrc (proxy, https-proxy). The extension of the respective axios calls in teams.js or telegram.js with the proxy address (https://www.npmjs.com/package/axios > `axios#post(url[, data[, config]])`) also fails because the request doesn't arrive at the Squid with `request_method=CONNECT`.
The webhooks for Telegram and teams via the squid work as desired via curl. 

### ✔️ Solution

The general use of the system variables (`HTTP_PROXY`, `HTTPS_PROXY`, `NO_PROXY`) for the installation (`download-dist.js`), the actual monitoring (already works via the `http-proxy-agent` module) and the notifications (notification-providers) would make sense. 

### ❓ Alternatives

_No response_

### 📝 Additional Context

I was able to get the download to run as part of the installation using the 'https-proxy-agent' module in download-dist.js. I couldn't get Axios to work with the Squid.

### PR DESCRIPTION
**Pull request by koehdaniel**
As explained in Issue #2588 the Teams-Notification currently does not use the system proxy. By using request instead of axios this is fixed.

⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [X] I have read and understand the pull request rules.

# Description

Partially addresses #2588 

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [X] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Comment by Zaid-maker** [comment]
Isn't request module deprecated?

File: server/notification-providers/teams.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const NotificationProvider = require("./notification-provider");
-const axios = require("axios");
+const request = require("request");
```
Comment (louislam) [inline server/notification-providers/teams.js:2]: As @Zaid-maker said, it is deprecated.

Comment (Zaid-maker) [reply server/notification-providers/teams.js:2]: Yes it's deprecated and have high dependency vulnerabilities issues. So it's adviced to use axios in place

**Comment by vanhoutenbos** [comment]
@koehdaniel will you update this PR? I would love this feature!

**Comment by louislam** [comment]
Trying to clean some staled pull requests, if you have found an alternative of `request`, feel free to re-open this pull request again. Thanks.

**Pull request closed without merging**